packages/backend/src/tokens/authenticateContext.ts

```diff
@@ -0,0 +1,88 @@
+import { constants } from '../constants';
+import type { ClerkRequest } from './clerkRequest';
+import type { AuthenticateRequestOptions } from './types';
+
+interface AuthenticateContextInterface extends AuthenticateRequestOptions {
+  // header-based values
+  sessionTokenInHeader: string | undefined;
```

Comment:
❓ why do we use `string | undefined` instead of `?: string`

Reply:
In general, the two variations are almost always interchangeable, with one semantic difference being that `string | undefined` would mean that the prop exists in the object but is undefined (aka `"something" in obj` would return `true`) while `?: string` means that it might now exist at all (`"something" in obj` might return false)

Here, we're initializing the values and even if some of them will be `undefined`, they do exist on `this`. Extremely nitpicky difference here so happy to change if you prefer the alternative syntax.